iocore/net/SSLUtils.cc

```diff
@@ -1225,6 +1233,17 @@ SSLMultiCertConfigLoader::init_server_ssl_ctx(std::vector<X509 *> &cert_list, co
 
     SSL_CTX_set_session_cache_mode(ctx, SSL_SESS_CACHE_SERVER | SSL_SESS_CACHE_NO_INTERNAL | additional_cache_flags);
 
+#if OPENSSL_1_1_1_API
+    // Must disable anti-replay if external cache is used with 0-rtt,
+    // otherwise session reuse will be broken. The freshness check
+    // described in https://tools.ietf.org/html/rfc8446#section-8.3
+    // is still performed so we are fine.
```

Comment:
Unfortunately, as RFC 8446 8.3.  Freshness Checks describes, we are not fine :(

> Note that freshness checking alone is not sufficient to prevent
   replays because it does not detect them during the error window,
   which -- depending on bandwidth and system capacity -- could include
   billions of replays in real-world settings.

As first step of 0-RTT support, we can go with freshness checking alone. But definitely we need to add protections.

Some ideas are accepting early data if 1). HTTP method is safe, 2). cache hit, or 3). origin server has Early-Data Header support. Anyway, we need to allow users to choose policy depends on their use cases.

Reply:
Hmm, I need to do more reading on this then.

Reply:
Agreed. We have to implement this properly to the specs: https://tools.ietf.org/html/rfc8470